apps/verify/behaviours/validate-case-details.js

```diff
@@ -1,21 +1,23 @@
+const axios = require('axios');
 const config = require('../../../config');
+const baseUrl = `${config.saveService.host}:${config.saveService.port}/saved_applications/cepr/`;
+const moment = require('moment');
 
-module.exports = superclass => class extends superclass {
-  saveValues(req, res, next) {
-    if (!this.isValidCase(req)) {
-      return res.redirect('/details-not-found');
-    }
-    return super.saveValues(req, res, next);
-  }
+module.exports = SuperClass => class extends SuperClass {
+  async validate(req, res, next) {
+    try {
+      const cepr = req.form.values.cepr;
+      const response = await axios.get(baseUrl + cepr);
+      const ceprRecords = response.data;
+      const dateOfBirth = ceprRecords.map(f => { return f.date_of_birth; });
 
-  isValidCase(req) {
-    const casesJson = require(`../../../data/${config.casesIds.S3Id}.json`);
-    const uan = req.form.values.uan;
-    const dob = req.form.values['date-of-birth'];
-
-    const imaResult = casesJson.find(
-      obj => obj.uan === uan && obj['date-of-birth'] === dob);
-    req.sessionModel.set('service', 'ima');
-    return imaResult;
+      if (!ceprRecords.length || (ceprRecords.length && dateOfBirth.toString() !==
+        moment(req.form.values['date-of-birth']).format('YYYY-MM-DD'))) {
+        return res.redirect('/details-not-found');
+      }
+    } catch (e) {
+      return next(e);
+    }
+    return super.validate(req, res, next);
```

Comment:
As discussed, sticking with validation from the JSON until the new table is ready to poll for correct values.

Reply:
Changed it back to use the JSON for validation.